nemo/collections/common/tokenizers/huggingface/auto_tokenizer.py

```diff
@@ -95,39 +95,48 @@ def __init__(
             cls_token: class token. Usually equal to bos_token
             unk_token: token to use for unknown tokens
         """
-
-        if pretrained_model_name not in ALL_PRETRAINED_CONFIG_ARCHIVE_MAP:
-            raise ValueError(f"{pretrained_model_name} not a huggingface pretrained model")
-
-        if vocab_file is not None:
-            self.tokenizer = AUTOTOKENIZER.from_pretrained(
-                pretrained_model_name_or_path=pretrained_model_name, vocab_file=vocab_file
-            )
-        else:
-            self.tokenizer = AUTOTOKENIZER.from_pretrained(pretrained_model_name_or_path=pretrained_model_name)
-
-        self.tokenizer.eos_token = self.tokenizer.sep_token
-        self.tokenizer.bos_token = self.tokenizer.cls_token
+        try:
+            if vocab_file is not None:
+                self.tokenizer = AUTOTOKENIZER.from_pretrained(
+                    pretrained_model_name_or_path=pretrained_model_name, vocab_file=vocab_file
+                )
+            else:
+                self.tokenizer = AUTOTOKENIZER.from_pretrained(pretrained_model_name_or_path=pretrained_model_name)
+        except Exception as e:
+            raise ValueError(f'{pretrained_model_name} is not supported by HuggingFace. {e}')
 
         special_tokens_dict = {}
-        if unk_token:
+        if self.tokenizer.unk_token is None:
             special_tokens_dict["unk_token"] = unk_token
-        if sep_token:
```

Comment:
@ekmb just realized that now you are not able to set your own custom special tokens anymore please fix that, thanks

Reply:
@yzhang123 what exactly do you want me to fix? you wanted to use eos_token for sep_token when available, no?  
Users can add any special tokens by calling add_special_tokens() and including "'additional_special_tokens'" to the dict of the special tokens after the tokenizer is initialized - the same way it's done in HF.

Reply:
@ekmb if you pass now special tokens to the constructor it wont work, that is not ok. The constructor should always work
e.g. if you pass sep_token="<<<sep>>>" it won't be updated if the tokenizer.tokenizer.sep_token is set